File: VB/MySolution.Module.Blazor/ButtonModel.vb
Imports System
Imports DevExpress.ExpressApp.Blazor.Components.Models

Namespace MySolution.Module.Blazor
	Public Class ButtonModel
		Inherits ComponentModelBase

		Public Property Text() As String
			Get
				Return GetPropertyValue(Of String)()
			End Get
			Set(ByVal value As String)
				SetPropertyValue(value)
			End Set
		End Property
		Public Sub ClickFromUI()
			ClickEvent?.Invoke(Me, EventArgs.Empty)
		End Sub
		Public Event Click As EventHandler
	End Class
End Namespace

File: CS/MySolution.Module.Blazor/ButtonModel.cs
using System;
using DevExpress.ExpressApp.Blazor.Components.Models;

namespace MySolution.Module.Blazor {
    public class ButtonModel : ComponentModelBase {
        public string Text {
            get => GetPropertyValue<string>();
            set => SetPropertyValue(value);
        }
        public void ClickFromUI() {
            Click?.Invoke(this, EventArgs.Empty);
        }
        public event EventHandler Click;
    }
}
